internal/rebac_admin/resources.go

```diff
@@ -30,7 +32,15 @@ func (s *resourcesService) ListResources(ctx context.Context, params *resources.
 		return nil, err
 	}
 	currentPage, expectedPageSize, pagination := pagination.CreatePaginationWithoutTotal(params.Size, params.Page)
-	res, err := s.jimm.ListResources(ctx, user, pagination)
+	namePrefixFilter, typeFilter := utils.GetNameAndTypeResourceFilter(params.EntityName, params.EntityType)
+	if typeFilter != "" {
+		typeFilter, err = validateAndConvertResourceFilter(typeFilter)
+		if err != nil {
+			return nil, v1.NewInvalidRequestError(err.Error())
+		}
+	}
```

Comment:
For a follow-up, but I think this should be calling an application layer function to create custom types. Like the idea I was suggesting earlier, the API layer can pass the strings it has received from the client into application layer validators that return app layer types that can be used in ListResources().

Reply:
Because a string doesn't tell me much, but a `type resource struct{ resource string }` tells me a lot more.